### PR TITLE
LOOP-3057 Updates to insertion and finish screens

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-3057

* Update UI elements and text copy to match design
* Allow rescheduling of expiration reminder on finish screen
* Add PodUIDemo app for more rapid iteration on Pod UI.
* Handle recoverable/non-recoverable errors appropriately.

### CODE REVIEW

File: DashKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1490,6 +1616,48 @@
...
 			name = Release;
 		};
+		C1B1CC1B25EE889E004D05C4 /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = 75U4X84TEG;
```
Comment (darinkrauss) [inline DashKit.xcodeproj/project.pbxproj]: Consider `DEVELOPMENT_TEAM = "";` for consistency with other targets in this project.

File: DashKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1490,6 +1616,48 @@
...
+			name = Debug;
+		};
+		C1B1CC1C25EE889E004D05C4 /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = 75U4X84TEG;
```
Comment (darinkrauss) [inline DashKit.xcodeproj/project.pbxproj]: Consider `DEVELOPMENT_TEAM = "";` for consistency with other targets in this project.

File: DashKit/DashPumpManager.swift [hunk 3]
```diff
@@ -629,6 +633,16 @@ open class DashPumpManager: PumpManager {
         }
     }
     
+    public var allowedExpirationReminderDateRange: ClosedRange<Date>? {
+        if let expiration = podExpiresAt {
```
Comment (darinkrauss) [inline DashKit/DashPumpManager.swift]: [nit] Consider `guard let expiration = podExpiresAt else { return nil }` form.

File: DashKit/Mocks/MockPodCommManager.swift [hunk 4]
```diff
@@ -194,6 +194,14 @@ public class MockPodCommManager: PodCommManagerProtocol {
                 eventListener(.error(error))
             }
         } else {
+            
+            
+            DispatchQueue.main.asyncAfter(deadline: .now() + 11) {
+                eventListener(.error(.bleCommunicationError))
+            }
+            return
```
Comment (darinkrauss) [inline DashKit/Mocks/MockPodCommManager.swift]: Erroneous `return` here?

**Review by rickpasetto — APPROVED** [review]
Are there any unit tests you'd like to add for these changes?  Otherwise, LGTM!

**Review by darinkrauss — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**